tgui/packages/tgui/interfaces/LocalGenerator/index.tsx

```diff
@@ -0,0 +1,186 @@
+
+import { useBackend } from '../../backend';
+import { Window } from '../../layouts';
+import { Section, Stack, Button, Box, ProgressBar } from '../../components';
+import { PortableHoldingTank } from '../common/PortableAtmos';
+import { LocalGeneratorData } from './type';
+
+export const LocalGenerator = (props, context) => {
+  const { act, data } = useBackend<LocalGeneratorData>(context);
+
+  const {
+    name,
+    holding,
+    internalCell,
+    connectedAPC,
+    chargeAPC,
+    boltsStatus,
+    generatorStatus,
+  } = data;
+
+  const handleToggleBolts = () => {
+    act("toggle-bolts");
+  };
+
+  const handleToggleGenerator = () => {
+    act("toggle-generator");
+  };
+
+  const handleSwapChargeTarget = () => {
+    act("swap-target");
+  };
+
+  const handleEjectTank = () => {
+    act("eject-tank");
+  };
+
+  const handleEjectCell = () => {
+    act("eject-cell");
+  };
+
+  const handleConnectAPC = () => {
+    act("connect-APC");
+  };
```

Comment:
I'm not requiring the change, as having defined handlers is broadly useful for abstracting away behavior into a named "doWhatever" function, but in this case the functions are making it harder to really determine what's going on when they're used, and `onClick={() => act('toggle-bolts')}` is no less performant or whatever.